counterfit/targets/cart_pole/DCPW.py

```diff
@@ -0,0 +1,369 @@
+# adapted from https://pytorch.org/tutorials/intermediate/reinforcement_q_learning.html
+
+import torch
+import torch.nn as nn
+import torch.optim as optim
+import torch.nn.functional as F
+import torchvision.transforms as T
+import tensorflow as tf
+
+import gym
+
+import math
+import random
+import numpy as np
+import matplotlib
+import matplotlib.pyplot as plt
+from collections import namedtuple, deque
+from itertools import count
+from PIL import Image
+import pickle
+import inspect
+import os
+import pathlib
+
+BATCH_SIZE = 128
+GAMMA = 0.999
+EPS_START = 0.9
+EPS_END = 0.05
+EPS_DECAY = 200
+TARGET_UPDATE = 10
+device = torch.device('cpu') # torch.device("cuda" if torch.cuda.is_available() else "cpu")
+global Transition 
```

Comment:
Declaring `Transition` as a global variable is not needed since this variable is global by default when you declared outside of the class/functions so it can be removed.